prow/monitoring/alertmanager.yaml

```diff
@@ -0,0 +1,20 @@
+global:
+  resolve_timeout: 5m
+
+route:
+  group_by: ['job']
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 2h
+  receiver: 'slack-notifications'
+
+receivers:
+  - name: 'slack-notifications'
+    slack_configs:
+      - channel: '#testing-ops'
+        api_url: '{{ api_url }}'
```

Comment:
This is a secret value

Reply:
@cjwagner

Reply:
in our production ... it looks like

```
api_url: 'https://hooks.slack.com/services/xxx'
```

That is url for the [slack incoming webhook](https://api.slack.com/incoming-webhooks).

Reply:
Why is the API URL a secret? How is this value populated?

Reply:
Do these incoming webhooks (https://api.slack.com/incoming-webhooks) allow anyone to post comments if they know the URL? Is that secure?

I was expecting a slack token to be used for authentication like Prow's Slack client uses: https://github.com/kubernetes/test-infra/blob/f23ec350fe0fbcf3da8cf99529dfcc32c577f9bd/prow/cmd/hook/main.go#L89

Reply:
> Why is the API URL a secret? How is this value populated?

Knowing the url, one can send messages to any slack channels of the workspace the account belongs to.

That was one of my questions in the originl PR:
How do we handle sensitive information in CI of prow cluster?
In our CI, all the secrets get populated from [bitwarden](https://bitwarden.com/).

Reply:
If you have that URL you can post to Slack. In our deployment we're doing something suboptimal -- 

```
sed -e "s|{{ api_url }}|$(slack_api_url)|g" ./alertmanager.yaml > /tmp/alertmanager_with_secret.yaml
kubectl create -n prow-monitoring secret generic alertmanager-prow --from-file=alertmanager.yaml=/tmp/alertmanager_with_secret.yaml --from-file=./msg.tmpl --dry-run -o yaml | kubectl apply -f -
```

Hopefully `k8s_secret` would let us do the injection in a better way?

Reply:
> Hopefully `k8s_secret` would let us do the injection in a better way?

`k8s_secret` provides `kubectl create secret --frome-file` while `k8s_object` works as `kubectl apply -f `.

How we inject the secret value ... it is another story though. ^_^